test/remote/account_unlock_tests.js

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var test = require('../ptaptest')
+var url = require('url')
+var Client = require('../client')
+var TestServer = require('../test_server')
+
+
+var config = require('../../config').root()
+
+TestServer.start(config)
+.then(function main(server) {
+
+  test(
+    'unlock account',
+    function (t) {
+      var email = server.uniqueEmail()
+      var password = 'allyourbasearebelongtous'
+      var client = null
+      return Client.createAndVerify(config.publicUrl, email, password, server.mailbox)
+      .then(
+        function () {
+          return Client.login(config.publicUrl, email, password)
+        }
+      )
+      .then(
+        function (x) {
+          client = x
+        }
+      )
+      .then(
+        function () {
+          return client.lockAccount(email, password)
+        }
+      )
+      .then(
+        function () {
+          return client.resendAccountUnlockCode()
+        }
+      )
+      .then(
+        function () {
+          return server.mailbox.waitForCode(email)
+        }
+      )
+      .then(
+        function (code) {
+          return client.verifyAccountUnlockCode(client.uid, code)
+        }
+      )
+    }
+  )
+
```

Comment:
Just for clarification, is this test just making sure that all of these calls are successful (and relies on `ptaptest` to complain if we hit an error)?